_inc/connect-button.js

```diff
@@ -1,6 +1,57 @@
+/* global jpConnect */
+
 jQuery( document ).ready( function( $ ) {
 	$( '.jp-connect-button' ).click( function( event ) {
 		event.preventDefault();
-		alert( '❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️' );
+		if ( ! jetpackConnectButton.isRegistering ) {
+			jetpackConnectButton.handleClick();
+		}
 	} );
+
+	var jetpackConnectButton = {
+		isRegistering: false,
+		handleClick: function() {
+			jetpackConnectButton.isRegistering = true;
+			$( '.jp-connect-button' ).text( jpConnect.buttonTextRegistering );
+			console.log( 'sending request to', jpConnect.apiBaseUrl + '/connection/register' );
+			$.ajax( {
+				url: jpConnect.apiBaseUrl + '/connection/register',
+				type: 'POST',
+				data: {
+					registration_nonce: jpConnect.registrationNonce,
+					_wpnonce: jpConnect.apiNonce,
+				},
+				error: function( error ) {
+					console.log( 'request failed' );
+					console.log( error );
+					jetpackConnectButton.isRegistering = false;
+				},
+				success: function( data ) {
+					console.log( 'request success' );
+					window.addEventListener( 'message', jetpackConnectButton.receiveData );
+					$( '.jp-connect-full__button-container' ).html(
+						'<iframe src="' + data.authorizeUrl + '" class="jp-jetpack-connect__iframe" />'
+					);
+				},
+			} );
+		},
+		receiveData: function( event ) {
+			if ( event.origin === 'https://jetpack.wordpress.com' ) {
```

Comment:
should we use APIBASE or something alike?

Reply:
yes we should. that's likely why @tyxla was not able to redirect during his flow

Reply:
Nice catch!